test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/SelectExpandClauseExtensionsTests.cs

```diff
@@ -0,0 +1,280 @@
+        {
+            //Arrange
+            var expected = new List<string>();
+            expected.Add("FavoriteDate");
+            expected.Add("Name");
+            expected.Add("MyAddress");
+            expected.Add("MyAddress/MyNeighbors/City");
+
+            //Act
+            var clause = CreateSelectExpandClauseNestedSelectWithNestedType();
+            var actual = SelectExpandClauseExtensions.GetCurrentLevelSelectList(clause);
+
+            //Assert
+            Assert.True(expected.SequenceEqual(actual));
+        }
+
+        [Fact]
+        public void GetCurrentLevelSelectListTestNestedSelectNestedTypeWithComplex()
+        {
+            //Arrange
+            var expected = new List<string>();
+            expected.Add("FavoriteDate");
+            expected.Add("Name");
+            expected.Add("MyAddress");
+
+            //Act
+            var clause = CreateSelectExpandClauseNestedSelectWithNestedAndComplex();
+            var actual = SelectExpandClauseExtensions.GetCurrentLevelSelectList(clause);
+
+            //Assert
+            Assert.True(expected.SequenceEqual(actual));
+        }
+
+        [Fact]
+        public void GetCurrentLevelSelectListTestAllSelected()
+        {
+            //Arrange
+            var expected = new List<string>();
+            expected.Add("*");
+            
+            //Act
+            var clause = CreateSelectExpandClauseAllSelected();
+            var actual = SelectExpandClauseExtensions.GetCurrentLevelSelectList(clause);
+
+            //Assert
+            Assert.Equal(expected.Count, actual.Count);
+            Assert.Equal(expected[0], actual[0]);            
+        }
+
+        [Fact]
+        public void GetCurrentLevelSelectListTestAllSelectedWithNamespace()
+        {
+            //Arrange
+            var expected = new List<string>();
+            expected.Add("namespace.*");
+
+            //Act
+            var clause = CreateSelectExpandClauseAllSelectedWithNamespace();
+            var actual = SelectExpandClauseExtensions.GetCurrentLevelSelectList(clause);
+
+            //Assert
+            Assert.Equal(expected.Count, actual.Count);
+            Assert.Equal(expected[0], actual[0]);
+        }
+
+        private static SelectExpandClause CreateSelectExpandClauseMultiSelect()
+        {
+            ODataSelectPath personNamePath = new ODataSelectPath(new PropertySegment(HardCodedTestModel.GetPersonNameProp()));
+            ODataSelectPath personShoePath = new ODataSelectPath(new PropertySegment(HardCodedTestModel.GetPersonShoeProp()));
+
+            var clause = new SelectExpandClause(new List<SelectItem>(), false);
+            clause.AddToSelectedItems(new PathSelectItem(personShoePath));
+            clause.AddToSelectedItems(new PathSelectItem(personNamePath));
+
+            return clause;
+        }
+
+        private static SelectExpandClause CreateSelectExpandClauseNestedSelect()
+        {

```

Comment:
It would be helpful for each of these to show (i.e., in a comment) the select/expand that was being built. As it is, it's difficult to read through the code to try and figure out what scenarios are being tested.

Reply:
#Resolved